### PR TITLE
Check ``getTlsTrustStorePath`` NPE when user forget to set it.

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

If a user sets ``useKeyStoreTls=true`` then forget to set ``getTlsTrustStorePath``, we’re having NPE which becomes hard to debug for users.

### Modifications

- Add NPE check and give use more clear error information.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java [hunk 1]
```diff
@@ -72,7 +72,10 @@ public PulsarChannelInitializer(ClientConfigurationData conf, Supplier<ClientCnx
         if (tlsEnabled) {
             if (tlsEnabledWithKeyStore) {
                 AuthenticationDataProvider authData1 = conf.getAuthentication().getAuthData();
-
+                if (conf.getTlsTrustStorePath() == null) {
+                    throw new RuntimeException("Failed to create TLS context, Due to empty "
```
Comment (Jason918) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java:76]:
```suggestion
                    throw new RuntimeException("Failed to create TLS context, due to empty "
```

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java [hunk 2]
```diff
@@ -72,7 +72,10 @@ public PulsarChannelInitializer(ClientConfigurationData conf, Supplier<ClientCnx
         if (tlsEnabled) {
             if (tlsEnabledWithKeyStore) {
                 AuthenticationDataProvider authData1 = conf.getAuthentication().getAuthData();
-
+                if (conf.getTlsTrustStorePath() == null) {
```
Comment (codelipenghui) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java:75]: We should check if it is an empty String? Empty string also does not make any sense here.

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java (hunk 1 above)
Comment (codelipenghui) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java:76]: We should throw PulsarClientException?

Comment (codelipenghui) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java:76]: I think it's better to use like "Failed to create TLS context, the tlsTrustStorePath need to be configured if useKeyStoreTls enabled" ?

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java (hunk 2 above)
Comment (mattisonchao) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java]: fixed.

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java (hunk 1 above)
Comment (mattisonchao) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java]: fixed

Comment (mattisonchao) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarChannelInitializer.java]: Hi, @Jason918.
I agree with @codelipenghui.

**Comment by mattisonchao** [comment]
@codelipenghui  @Jason918  PTAL :)

**Review by merlimat — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]

**Review by Shoothzj — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Hi @momo-jun can you update docs for this PR? Thanks!
Context: https://streamnative.slack.com/archives/CSEST9M51/p1644604343198309

**Comment by momo-jun** [comment]
> Hi @momo-jun can you update docs for this PR? Thanks! Context: https://streamnative.slack.com/archives/CSEST9M51/p1644604343198309

I've submitted the doc PR. PTAL.

**Pull request merged**